moveit_ros/planning/planning_request_adapter_plugins/src/chomp_optimizer_adapter.cpp

```diff
@@ -0,0 +1,242 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2012, Willow Garage, Inc.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of Willow Garage nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Raghavender Sahdev */
+
+#include <moveit/planning_request_adapter/planning_request_adapter.h>
+#include <moveit/robot_state/conversions.h>
+#include <moveit/trajectory_processing/trajectory_tools.h>
+#include <moveit_msgs/RobotTrajectory.h>
+#include <class_loader/class_loader.hpp>
+#include <ros/ros.h>
+
+#include <chomp_motion_planner/chomp_planner.h>
+#include <chomp_motion_planner/chomp_parameters.h>
+#include <moveit/planning_interface/planning_interface.h>
+#include <moveit/trajectory_processing/iterative_time_parameterization.h>
+
+#include <moveit/collision_distance_field/collision_detector_allocator_hybrid.h>
+
+#include <tf/transform_listener.h>
+
+#include <moveit/robot_state/conversions.h>
+
+#include <vector>
+#include <eigen3/Eigen/Core>
+
+using namespace chomp;
+
+namespace default_planner_request_adapters
+{
+class CHOMPOptimizerAdapter : public planning_request_adapter::PlanningRequestAdapter
+{
+public:
+  ChompPlanner chomp_interface_;
+  moveit::core::RobotModelConstPtr robot_model_;
+
+  chomp::ChompParameters params_;
+  boost::shared_ptr<tf::TransformListener> tf_;
```

Comment:
move member functions to bottom of file / class

Reply:
done